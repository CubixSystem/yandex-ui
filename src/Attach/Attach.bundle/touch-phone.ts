import { withRegistry, Registry } from '@bem-react/di';
import { compose, composeU } from '@bem-react/core';

// base
import { Attach as AttachTouchPhone, cnAttach } from '../Attach@touch-phone';
// _size
import { withSizeL } from '../_size/Attach_size_l';
import { withSizeM } from '../_size/Attach_size_m';
import { withSizeS } from '../_size/Attach_size_s';

// Registry dependencies â€” Button
import { Button as ButtonDesktop } from '../../Button/Button@touch-phone';
import { withSizeL as withButtonSizeL } from '../../Button/_size/Button_size_l';
import { withSizeM as withButtonSizeM } from '../../Button/_size/Button_size_m';
import { withSizeS as withButtonSizeS } from '../../Button/_size/Button_size_s';
import { withViewDefault as withButtonViewDefault } from '../../Button/_view/Button_view_default';
import { withThemeNormal as withButtonThemeNormal } from '../../Button/_theme/Button_theme_normal';

const registry = new Registry({ id: cnAttach() });

const Button = compose(
    withButtonThemeNormal,
    withButtonViewDefault,
    composeU(withButtonSizeL, withButtonSizeM, withButtonSizeS),
)(ButtonDesktop);

registry.set('Button', Button);

export * from '../Attach@touch-phone';

export const Attach = compose(
    composeU(withSizeL, withSizeM, withSizeS),
    withRegistry(registry),
)(AttachTouchPhone);
