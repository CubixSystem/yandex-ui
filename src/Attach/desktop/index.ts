export * from '../Attach@desktop';
export * from '../_size/Attach_size_l';
export * from '../_size/Attach_size_m';
export * from '../_size/Attach_size_s';
