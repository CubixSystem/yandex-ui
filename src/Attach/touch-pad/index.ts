export * from '../Attach@touch-pad';
export * from '../_size/Attach_size_l';
export * from '../_size/Attach_size_m';
export * from '../_size/Attach_size_s';
