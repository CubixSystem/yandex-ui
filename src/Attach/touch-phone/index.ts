export * from '../Attach@touch-phone';
export * from '../_size/Attach_size_l';
export * from '../_size/Attach_size_m';
export * from '../_size/Attach_size_s';
