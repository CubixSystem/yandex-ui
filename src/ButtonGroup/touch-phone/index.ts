import { compose, composeU, ExtractProps } from '@bem-react/core';

import { ButtonGroup as ButtonGroupDesktop } from '../ButtonGroup';

// _margin
import { withGapS } from '../_gap/ButtonGroup_gap_s';
import { withGapM } from '../_gap/ButtonGroup_gap_m';
import { withGapL } from '../_gap/ButtonGroup_gap_l';
import { withGapXL } from '../_gap/ButtonGroup_gap_xl';
// _pin
import { withPinCircle } from '../_pin/ButtonGroup_pin_circle';
import { withPinRound } from '../_pin/ButtonGroup_pin_round';

export * from '../ButtonGroup';
export * from '../ButtonGroup.utils';

export const ButtonGroup = compose(
    composeU(withGapS, withGapM, withGapL, withGapXL),
    composeU(withPinCircle, withPinRound),
)(ButtonGroupDesktop);

export type ButtonGroupProps = ExtractProps<typeof ButtonGroup>;
