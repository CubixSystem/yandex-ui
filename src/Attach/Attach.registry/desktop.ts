import { Registry } from '@bem-react/di';

import { cnAttach } from '../Attach.const';
import { Holder } from '../Holder/Attach-Holder';
import { Control } from '../Control/Attach-Control';

export const attachRegistry = new Registry({ id: cnAttach() });

attachRegistry
    .set('Holder', Holder)
    .set('Control', Control)
    .set('Button', () => {
        // TODO: Использовать error api из новой версии react.
        throw new Error(`Для работы "Attach" требуется установить компонент "Button" в реестр "${cnAttach()}".`);
    });
