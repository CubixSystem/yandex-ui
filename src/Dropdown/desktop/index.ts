import { ExtractProps } from '@bem-react/core';
import { withDropdown } from '../Dropdown';
import { Popup } from '../../Popup/desktop/bundle';

export const Dropdown = withDropdown<ExtractProps<typeof Popup>>(Popup);

export * from '../Dropdown';
