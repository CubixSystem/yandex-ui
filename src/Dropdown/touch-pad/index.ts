import { ExtractProps } from '@bem-react/core';
import { ComponentType } from 'react';
import { withDropdown, DropdownProps } from '../Dropdown';
import { Popup } from '../../Popup/Popup.bundle/touch-pad';

export const Dropdown = withDropdown<ExtractProps<typeof Popup>>(Popup) as ComponentType<DropdownProps>;
export * from '../Dropdown';
