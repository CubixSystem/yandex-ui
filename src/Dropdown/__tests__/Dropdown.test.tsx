import React from 'react';

import { createClientRender, screen, fireEvent } from '../../internal/testing';
import { Dropdown as DropdownDesktop } from '../desktop';
import { Dropdown as DropdownTouchPhone } from '../touch-phone';
import { Dropdown as DropdownTouchPad } from '../touch-pad';

const platforms = [
    ['desktop', DropdownDesktop],
    ['touch-phone', DropdownTouchPhone],
    ['touch-pad', DropdownTouchPad],
] as const;

describe.each(platforms)('Dropdown@%s', (_platform, Dropdown) => {
    const render = createClientRender();

    afterEach(() => {
        jest.useRealTimers();
    });

    test('должен вернуть полный шаблон компонента (snapshot)', () => {
        const { container } = render(
            <Dropdown content={<div>Test</div>}>
                <button />
            </Dropdown>,
        );

        expect(container.firstChild).toMatchSnapshot();
    });

    test('должен рендерить содержимое', () => {
        render(
            <Dropdown content={<div>Text</div>}>
                <button />
            </Dropdown>,
        );

        expect(screen.getByRole('button')).toBeInTheDocument();
    });

    test('не должен рендерить невидимый попап', () => {
        render(
            <Dropdown content={<div>Text</div>} data-testid="popup">
                <button />
            </Dropdown>,
        );

        expect(screen.queryByTestId('popup')).not.toBeInTheDocument();
    });

    test('должен рендерить видимый попап', () => {
        render(
            <Dropdown visible content={<div>Text</div>} data-testid="popup">
                <button />
            </Dropdown>,
        );

        expect(screen.getByTestId('popup')).toBeInTheDocument();
    });

    test('должен вызывать `onVisibleChange` c правильными аргументами по клику', () => {
        const visibleChangeHandler = jest.fn();
        render(
            <Dropdown onVisibleChange={visibleChangeHandler} trigger="click" content={<div>Text</div>}>
                <button />
            </Dropdown>,
        );

        fireEvent.click(screen.getByRole('button'));

        expect(visibleChangeHandler).toHaveBeenCalledTimes(1);
        expect(visibleChangeHandler).toHaveBeenCalledWith(true);

        fireEvent.click(screen.getByRole('button'));

        expect(visibleChangeHandler).toHaveBeenCalledTimes(2);
        expect(visibleChangeHandler).toHaveBeenCalledWith(false);
    });

    test('должен вызывать `onVisibleChange` c правильными аргументами по наведению', () => {
        const visibleChangeHandler = jest.fn();
        jest.useFakeTimers();

        render(
            <Dropdown onVisibleChange={visibleChangeHandler} content={<div>Text</div>}>
                <button />
            </Dropdown>,
        );

        fireEvent.mouseEnter(screen.getByRole('button'));
        jest.runOnlyPendingTimers();

        expect(visibleChangeHandler).toHaveBeenCalledTimes(1);
        expect(visibleChangeHandler).toHaveBeenCalledWith(true);

        fireEvent.mouseLeave(screen.getByRole('button'));
        jest.runOnlyPendingTimers();

        expect(visibleChangeHandler).toHaveBeenCalledTimes(2);
        expect(visibleChangeHandler).toHaveBeenCalledWith(false);
    });

    test('должен вызывать `onVisibleChange` c правильными аргументами по фокусу', () => {
        const visibleChangeHandler = jest.fn();
        jest.useFakeTimers();

        render(
            <Dropdown onVisibleChange={visibleChangeHandler} trigger="focus" content={<div>Text</div>}>
                <button>Dropdown</button>
            </Dropdown>,
        );

        fireEvent.focus(screen.getByRole('button'));
        jest.runOnlyPendingTimers();

        expect(visibleChangeHandler).toHaveBeenCalledTimes(1);
        expect(visibleChangeHandler).toHaveBeenCalledWith(true);

        fireEvent.blur(screen.getByRole('button'));
        jest.runOnlyPendingTimers();

        expect(visibleChangeHandler).toHaveBeenCalledTimes(2);
        expect(visibleChangeHandler).toHaveBeenCalledWith(false);
    });

    test('должен вызывать колбэк переданный в кнопке', () => {
        const handler = jest.fn();
        render(
            <Dropdown trigger={['click', 'hover']} content={<div>Text</div>}>
                <button onMouseLeave={handler} onClick={handler}>
                    Dropdown
                </button>
            </Dropdown>,
        );

        fireEvent.click(screen.getByRole('button'));
        expect(handler).toHaveBeenCalledTimes(1);

        fireEvent.mouseLeave(screen.getByRole('button'));
        expect(handler).toHaveBeenCalledTimes(2);
    });
});
