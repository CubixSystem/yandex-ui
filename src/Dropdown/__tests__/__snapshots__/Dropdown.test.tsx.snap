// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown@desktop должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Dropdown"
>
  <button />
</span>
`;

exports[`Dropdown@touch-pad должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Dropdown"
>
  <button />
</span>
`;

exports[`Dropdown@touch-phone должен вернуть полный шаблон компонента (snapshot) 1`] = `
<span
  class="Dropdown"
>
  <button />
</span>
`;
